doku/tests/printers.rs

```diff
@@ -87,9 +87,30 @@ mod prelude {
             doku::to_json_fmt::<$ty>(&fmt)
         }};
 
+        (@assert to_json_without_key_quotes($ty:ty)) => {{
```

Comment:
Could you move it next to `to_json()` (and similarly for rest)? -- So that the macro matchers read nicely:
- `to_json(...)`,
- `to_json_without_key_quotes(...)`,
- `to_json_fmt(...)`,
- `to_json_fmt_val(...)`,
- `to_json_val(...)`,
- `to_json_val_without_key_quotes(...)`.

Also, currently when you want to specify more than just `surround_keys_with_quotes = false;`, one has to fall back to `to_json_fmt` & friends - would it be possible to extend the macro a bit, so that we have:
```
(@assert to_json_without_key_quotes($ty:ty $(, $fmt:tt)?)) => {{
    let mut fmt = /* from $fmt if present, default otherwise */;
    fmt.objects_style.surround_keys_with_quotes = false;

    /* ... */
}};
```
This would allow us to change e.g.:
```rust
"output.without_key_quotes.json" => to_json_fmt(Ty, {
    "objects_style": { "surround_keys_with_quotes": false },
    "enums_style": "Commented",
}),
```
... into:
```rust
"output.without_key_quotes.json" => to_json_fmt_without_key_quotes(Ty, {
    "enums_style": "Commented",
}),
```
.... which would look a bit better considering the context, I think:
```rust
printer_test! {
    "output.json" => to_json_fmt(Ty, {
         "enums_style": "Commented",
    }),

    "output.without_key_quotes.json" => to_json_fmt_without_key_quotes(Ty, {
        "enums_style": "Commented",
    }),
}
```